fea-rs/src/compile/tables/gdef.rs

```diff
@@ -40,17 +43,26 @@ pub struct GdefBuilder {
 
 impl GdefBuilder {
     pub fn build(&self) -> (tables::gdef::Gdef, Option<VariationIndexRemapping>) {
+        let mut var_store = self
+            .var_store
+            .clone()
+            // we need to always have a varstore builder in order to build the lig_caret
+            // table, but if we did not have one by now we aren't a variable font, so
+            // we will throw it away after.
+            .unwrap_or_else(|| VariationStoreBuilder::new(0));
         let mut table = tables::gdef::Gdef::new(
             self.build_class_def(),
             self.build_attach_list(),
-            self.build_lig_caret_list(),
+            self.build_lig_caret_list(&mut var_store),
             self.build_mark_attach_class_def(),
         );
 
         table.mark_glyph_sets_def = self.build_mark_glyph_sets().into();
-        if let Some((var_store, key_map)) = self.var_store.clone().map(VariationStoreBuilder::build)
-        {
+        // only use the var_store if we had one set at the start of this fn
+        let var_store = Some(var_store).filter(|_| self.var_store.is_some());
+        if let Some((var_store, key_map)) = var_store.map(VariationStoreBuilder::build) {
```

Comment:
Creating a mutable varstore that isn't self.var_store and then having to carefully remember if it's bogus or not seems fairly ugly. Why is that necessary? We can't or shouldn't code it to run against self.var_store and keep the original code here?

Reply:
At a glance it seems like it might be ok for build to take `&mut self`, as CompilationCtx::build - which appears to be the primary caller - does?

Reply:
agree that this is annoying, and is the consequence of a few things colliding:
- we need to mutate the var store to add deltas to it, in `build_lig_caret_list`.
- for Annoying Rust Reasons, `Option<&mut T>` is not `Copy`. This means you can't do something like,

```rust
let mut thing = Thing;
let thing_ref = Some(&mut Thing);
for _ in 0..5 {
    do_thing(thing_ref); // < error: `thing_ref` moved in previous loop pass
}
```
- and as a consequence of this, `build_lig_caret_list` (along with many other similar functions) take `&mut VariationStoreBuilder` instead of `Option<&mut VariationStoreBuilder>`
- and this means that even if there is no varstore at this point (and this is late in compilation, if this was a variable font we would have a varstore) we need to create one in order to call that method; but if we created one just for this we don't care to keep it on, and so we discard it afterwards.

having `&mut self` wouldn't really change anything; we would still need to create a dummy varstore in order to build the lig caret list, and we would still discard it afterwards.

Reply:
Per IM checking if the varstore is empty instead of `Some(var_store).filter(|_| self.var_store.is_some())` should suffice.